src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -77,28 +79,31 @@ protected Integer initialValue()
         }
     };
 
-
     public TrieMemoryIndex(IndexContext indexContext)
     {
         super(indexContext);
-        //TODO Do we need to follow a setting for this?
-        this.data = new MemtableTrie<>(BufferType.OFF_HEAP);
+        this.data = new MemtableTrie<>(TrieMemtable.BUFFER_TYPE);
         this.primaryKeysReducer = new PrimaryKeysReducer();
-        // MemoryIndex is per-core, so analyzer should be thread-safe..
-        this.analyzerFactory = indexContext.getAnalyzerFactory();
-        this.validator = indexContext.getValidator();
-        this.isLiteral = TypeUtil.isLiteral(validator);
     }
 
-    @Override
-    public long add(DecoratedKey key, Clustering clustering, ByteBuffer value)
+    public void add(DecoratedKey key,
+                    Clustering clustering,
+                    ByteBuffer value,
+                    LongConsumer onHeapAllocationsTracker,
+                    LongConsumer offHeapAllocationsTracker)
     {
-        synchronized (writeLock)
```

Comment:
Why this change? 
You don't seem to be using any of the additional functionality provided by the `ReentrantLock`, unless you really need fairness here. But generally I'd expect `ReentrantLock` to be slightly more complex internally and slower than a synchronized, because `synchronized` doesn't need to be fair (and it doesn't provide all the other features).